geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/services/StripedExecutorService.java

```diff
@@ -0,0 +1,522 @@
+   * within the allotted amount of time.
+   */
+  public boolean awaitTermination(long timeout, TimeUnit unit)
+      throws InterruptedException {
+    lock.lock();
+    try {
+      long waitUntil = System.nanoTime() + unit.toNanos(timeout);
+      long remainingTime;
+      while ((remainingTime = waitUntil - System.nanoTime()) > 0
+          && !executors.isEmpty()) {
+        terminating.awaitNanos(remainingTime);
+      }
+      if (remainingTime <= 0)
+        return false;
+      if (executors.isEmpty()) {
+        return executor.awaitTermination(
+            remainingTime, TimeUnit.NANOSECONDS);
+      }
+      return false;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  /**
+   * As soon as a SerialExecutor is empty, we remove it from the
+   * executors map. We might thus remove the SerialExecutors
+   * more quickly than necessary, but at least we can avoid a
+   * memory leak.
+   */
+  private void removeEmptySerialExecutor(Object stripe, SerialExecutor ser_ex) {
+    assert ser_ex == executors.get(stripe);
+    assert lock.isHeldByCurrentThread();
+    assert ser_ex.isEmpty();
+
+    executors.remove(stripe);
+    terminating.signalAll();
+    if (state == State.SHUTDOWN && executors.isEmpty()) {
+      executor.shutdown();
+    }
+  }
+
+  /**
+   * Prints information about current state of this executor, the
+   * wrapped executor and the serial executors.
+   */
+  public String toString() {
+    lock.lock();
+    try {
+      return "StripedExecutorService: state=" + state + ", " +
+          "executor=" + executor + ", " +
+          "serialExecutors=" + executors;
+    } finally {
+      lock.unlock();
+    }
+
+  }
+
+  /**
+   * This field is used for conditional compilation. If it is
+   * false, then the finalize method is an empty method, in
+   * which case the SerialExecutor will not be registered with
+   * the Finalizer.
+   */
+  private static boolean DEBUG = false;
+
+  /**
+   * SerialExecutor is based on the construct with the same name
+   * described in the {@link Executor} JavaDocs. The difference
+   * with our SerialExecutor is that it can be terminated. It
+   * also removes itself automatically once the queue is empty.
+   */
+  private class SerialExecutor implements Executor {
+    /**
+     * The queue of unexecuted tasks.
+     */
+    private final BlockingQueue<Runnable> tasks = new LinkedBlockingQueue<>();
+    /**
+     * The runnable that we are currently busy with.
+     */
+    private Runnable active;
+    /**
+     * The stripe that this SerialExecutor was defined for. It
+     * is needed so that we can remove this executor from the
+     * map once it is empty.
+     */
+    private final Object stripe;
+
+    /**
+     * Creates a SerialExecutor for a particular stripe.
+     */
+    private SerialExecutor(Object stripe) {
+      this.stripe = stripe;
+      if (DEBUG) {
+        System.out.println("SerialExecutor created " + stripe);
```

Comment:
Rather than the `DEBUG` boolean and a `System.out` statement, might it be better to modify this to make use of logging?

Reply:
Added a `debug` log statement